src/acl/external/time_quota/ext_time_quota_acl.cc

```diff
@@ -73,140 +72,71 @@ TDB_CONTEXT *db = nullptr;
  */
 static int pauseLength = 300;
 
-static FILE *logfile = stderr;
-static int tq_debug_enabled = false;
-
-static void open_log(const char *logfilename)
-{
-    logfile = fopen(logfilename, "a");
-    if ( logfile == NULL ) {
-        perror(logfilename);
-        logfile = stderr;
-    }
-}
-
-static void vlog(const char *level, const char *format, va_list args)
-{
-    time_t now = time(NULL);
-
-    fprintf(logfile, "%ld %s| %s: ", static_cast<long int>(now),
-            program_name, level);
-    vfprintf (logfile, format, args);
-    fflush(logfile);
-}
-
-static void log_debug(const char *format, ...)
-{
-    va_list args;
-
-    if ( tq_debug_enabled ) {
-        va_start (args, format);
-        vlog("DEBUG", format, args);
-        va_end (args);
-    }
-}
-
-static void log_info(const char *format, ...)
-{
-    va_list args;
-
-    va_start (args, format);
-    vlog("INFO", format, args);
-    va_end (args);
-}
-
-static void log_error(const char *format, ...)
-{
-    va_list args;
-
-    va_start (args, format);
-    vlog("ERROR", format, args);
-    va_end (args);
-}
-
-static void log_fatal(const char *format, ...)
+static void init_db(void)
 {
-    va_list args;
+    debugs(DEBUG_SECTION, 2, "opening time quota database \"" << db_path << "\".");
 
-    va_start (args, format);
-    vlog("FATAL", format, args);
-    va_end (args);
-}
+    int dbopts = TDB_CLEAR_IF_FIRST;
 
-static void init_db(void)
-{
-    log_info("opening time quota database \"%s\".\n", db_path);
-    db = tdb_open(db_path, 0, TDB_CLEAR_IF_FIRST, O_CREAT | O_RDWR, 0666);
+    db = tdb_open(db_path, 0, dbopts, O_CREAT | O_RDWR, 0666);
     if (!db) {
-        log_fatal("Failed to open time_quota db '%s'\n", db_path);
+        debugs(DEBUG_SECTION, DBG_IMPORTANT, "Failed to open time_quota db '" << db_path << '\'');
         exit(EXIT_FAILURE);
     }
+    // count the number of entries in the database, only used for debugging
+    debugs(DEBUG_SECTION, 2, "Database contains " << tdb_traverse(db, nullptr, nullptr) << " entries");
 }
 
 static void shutdown_db(void)
 {
     tdb_close(db);
 }
 
-static char *KeyString(int &len, const char *user_key, const char *sub_key)
+static SBuf KeyString(const char *user_key, const char *sub_key)
 {
-    static char keybuffer[TQ_BUFFERSIZE];
-    *keybuffer = 0;
-
-    len = snprintf(keybuffer, sizeof(keybuffer), "%s-%s", user_key, sub_key);
-    if (len < 0) {
-        log_error("Cannot add entry: %s-%s", user_key, sub_key);
-        len = 0;
-
-    } else if (static_cast<size_t>(len) >= sizeof(keybuffer)) {
-        log_error("key too long (%s,%s)\n", user_key, sub_key);
-        len = 0;
-    }
-
-    return keybuffer;
+    SBuf rv(user_key);
+    rv.append("-").append(sub_key);
+    return rv;
```

Comment:
Consider using this simpler and arguably more readable version instead:

```suggestion
    return ToSBuf(user_key, "-", sub_key);
```

I do not insist on this change.